### PR TITLE
Add yamltags

### PR DESCRIPTION
**Pull request by dlorenc**
This PR adds some new tags we can place on YAML structs that will make management of our config easier.

It adds a 'oneOf' tag, that can be placed on a set of fields and will ensure that only one of them is provided by a user.

It adds a 'required' tag, that can ensure the field has been set by the user.

It adds a 'default' tag, that can be used for strings and ints right now.

Getting this out early for comment. I'll keep cleaning it up and try integrating it into our config parsing next.

### CODE REVIEW

**Comment by dlorenc** [comment]
Got oneOf working and integrated into all our steps.

File: pkg/skaffold/yamltags/tags.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	switch val.Kind() {
+	case reflect.Int, reflect.Int16, reflect.Int32, reflect.Int64:
+		i, err := strconv.ParseInt(dt.dv, 0, 0)
+		if err != nil {
+			return err
+		}
+		val.SetInt(i)
+	case reflect.String:
+		val.SetString(dt.dv)
+	}
```
Comment (nkubala) [inline pkg/skaffold/yamltags/tags.go]: Is it possible that another type could get passed here? Maybe put a default case here that returns an error so it's more clear that other types aren't supported?

**Comment by dlorenc** [comment]
cc @nkubala I rebased this and it should be ready to go if we still want it.

File: examples/annotated-skaffold.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ build:
     gitCommit: {}
 
     # Tag the image with the checksum of the built image (image id).
-    sha256: {}
+    # sha256: {}
```
Comment (nkubala) [inline examples/annotated-skaffold.yaml]: did you mean to comment this out?

Comment (dlorenc) [reply examples/annotated-skaffold.yaml]: Yeah, this becomes invalid now. You can't define two tag policies with oneOf.

Comment (nkubala) [reply examples/annotated-skaffold.yaml]: Oh sorry, missed the `gitCommit` tagger above it.

**Pull request merged**